docs/03-features.md

```diff
@@ -0,0 +1,388 @@
+# Features
```

Comment:
Even md file needs a metadata block like the other ones in order to show up in the main README.

Reply:
Thanks, added them, please re-review. 
It would also be nice to have this documented somewhere (and this documentation ideally linked in the [README note](https://github.com/cloudfoundry/gorouter/blob/5174c184566bddac55a2fa42da4f67088de2c097/README.md?plain=1#L36-L40), rather than just referring to "a convention").